src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs

```diff
@@ -38,49 +40,66 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
+            var contract = ExtractBindingContract(parameter.ParameterType);
             if (contract == null)
             {
                 throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture,
-                    "Can't bind EventGridTriggerAttribute with publisher '{0}' to type '{1}'.", publisherName, parameter.ParameterType));
+                    "Can't bind EventGridTriggerAttribute to type '{0}'.", parameter.ParameterType));
             }
-            // unsupported publisher is caught in attribute constrcutor
-            return Task.FromResult<ITriggerBinding>(new EventGridTriggerBinding(context.Parameter, _extensionConfigProvider, context.Parameter.Member.Name, publisher, contract));
+
+            return Task.FromResult<ITriggerBinding>(new EventGridTriggerBinding(context.Parameter, _extensionConfigProvider, context.Parameter.Member.Name, contract));
 
         }
 
+        public Dictionary<string, Type> ExtractBindingContract(Type t)
+        {
+            if (t == typeof(EventGridEvent) || t == typeof(string))
+            {
+                var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+                // for javascript, 1st attempt is to return JSON string of EventGridEvent
+                contract.Add("EventGridTrigger", t);
+                contract.Add("data", typeof(JObject));
+                return contract;
+            }
+            else
+            {
+                return null;
+            }
+        }
 
         private class EventGridTriggerBinding : ITriggerBinding
         {
             private readonly ParameterInfo _parameter;
             private readonly IReadOnlyDictionary<string, Type> _bindingContract;
             private EventGridExtensionConfig _listenersStore;
             private readonly string _functionName;
-            private readonly IPublisher _publisher;
 
-            public EventGridTriggerBinding(ParameterInfo parameter, EventGridExtensionConfig listenersStore, string functionName, IPublisher publisher, Dictionary<string, Type> contract)
+            public EventGridTriggerBinding(ParameterInfo parameter, EventGridExtensionConfig listenersStore, string functionName, Dictionary<string, Type> contract)
             {
-                _publisher = publisher;
                 _listenersStore = listenersStore;
                 _parameter = parameter;
                 _functionName = functionName;
                 _bindingContract = contract;
             }
+            public Task<Dictionary<string, object>> ExtractBindingData(EventGridEvent e, Type t)
+            {
+                var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                if (t == typeof(EventGridEvent))
+                {
+                    bindingData.Add("EventGridTrigger", e);
+                }
+                else if (t == typeof(string))
```

Comment:
The binding data != parameters.  Separate concepts. The binding data is like auxiallary (language agnostic) information in addition to the parameter.